kyuubi-server/src/main/scala/org/apache/kyuubi/session/SessionLimiter.scala

```diff
@@ -101,14 +101,21 @@ class SessionLimiterImpl(userLimit: Int, ipAddressLimit: Int, userIpAddressLimit
   }
 }
 
-class SessionLimiterWithUnlimitedUsersImpl(
+class SessionLimiterWithAccessControlListImpl(
     userLimit: Int,
     ipAddressLimit: Int,
     userIpAddressLimit: Int,
-    var unlimitedUsers: Set[String])
+    var unlimitedUsers: Set[String],
+    var limitedUsers: Set[String])
   extends SessionLimiterImpl(userLimit, ipAddressLimit, userIpAddressLimit) {
   override def increment(userIpAddress: UserIpAddress): Unit = {
-    if (!unlimitedUsers.contains(userIpAddress.user)) {
+    val user = userIpAddress.user
+    if (StringUtils.isNotBlank(user) && limitedUsers.contains(user)) {
```

Comment:
I'm not familiar with the usage of SessionLimiter. Is this the right way or timing to do the rejection here?  cc @wForget

Reply:
I have the same question, the implementation looks incorrect, limited does not mean rejection.

Reply:
@pan3793 hi, I am new to kyuubi, could you give me more details about the meaning of `limited user`? From the issue  context I cannot get more input.

Reply:
I suppose when "limited user" is configured, connections from those users will be limited not rejected.

e.g. 

```
kyuubi.server.limit.connections.user.limited.list=tom,jerry
kyuubi.server.limit.connections.per.user=10
```

user "tom" and "jerry" are limited to opening up to 10 connections for each user, other users are not limited.

Reply:
1. that's why we discuss the priority of `limited.user.list` and `unlimited.user.list`, I think it's fine that `limited.user.list` takes high priority. (it means when `limited.user.list` is configured, ignore `unlimited.user.list`)
2. `user-ipaddress` should be affected by this change too. and we can introduce a similar change for `ipaddress` in another PR.